src/upload.js

```diff
@@ -50,7 +50,8 @@ SkynetClient.prototype.uploadDirectory = async function (directory, filename, cu
     formData.append(opts.portalDirectoryFileFieldname, ensureFileObjectConsistency(file), path);
   });
 
-  const url = makeUrl(this.portalUrl, opts.endpointPath, { filename });
+  let url = makeUrl(this.portalUrl, opts.endpointPath);
+  url = addUrlQuery(url, { filename });
```

Comment:
could you add query argument to makeUrl so it's all in one function?

Reply:
I don't think so, because `makeUrl` takes a variable number of arguments. Would be nice if it was possible, though.